dfttopif/parsers/vasp.py

```diff
@@ -261,14 +263,33 @@ def get_forces(self):
             conditions=Value(name="positions", vectors=self.atoms.positions.tolist())
         )
 
+    @staticmethod
+    def _get_bandgap_from_bands(energies, nelec):
+        """Compute difference in conduction band min and valence band max"""
+        nelec = int(nelec)
+        valence = [x[nelec-1] for x in energies]
+        conduction = [x[nelec] for x in energies]
+        return max(min(conduction) - max(valence), 0.0)
 
+    @staticmethod
+    def _get_bandgap_eigenval(eigenval_fname, outcar_fname):
+        """Get the bandgap from the EIGENVAL file"""
+        with open(outcar_fname, "r") as f:
+            parser = OutcarParser()
+            nelec = next(iter(filter(lambda x: "number of electrons" in x, parser.parse(f.readlines()))))["number of electrons"]
+        with open(eigenval_fname, "r") as f:
+            eigenval_info = list(EigenvalParser().parse(f.readlines()))
+        # spin_polarized = (2 == len(next(filter(lambda x: "kpoint" in x, eigenval_info))["occupancies"][0]))
+        # if spin_polarized:
+        all_energies = [zip(*x["energies"]) for x in eigenval_info if "energies" in x]
+        spin_energies = zip(*all_energies)
+        gaps = [VaspParser._get_bandgap_from_bands(x, nelec/2.0) for x in spin_energies]
```

Comment:
Does this mean that you are always assuming spin polarized?

Reply:
... since you divide by 2 to get the number of electrons

Reply:
Nope.  `x["energies"]` is always a list, with either one or two members.  There are always `nelec/2.0` occupied bands, since for non-polarized the bands each have a maximum occupation of `2.0`.